### ISSUE TITLE
[BUG] Regression Ensemble Model Does Not Handling Training Data Correctly

### ISSUE THREAD

**Issue by GitDoctor1996** [unit8co/darts#913]
**Describe the bug**
I have a simple univariate time series dataset. I am trying to use Regression Ensemble to come up with predictions that can sample from a wide variety of different approaches. Every time I create my list of models whether it be specifying the models on my own or just specifying the model type directly in the list with no hyperparameters specified, there are problems with the Auto_Arima and Prophet class in how it handles the training data and there are problems with the regression_train_n_points hyper parameter. 

To start off with regression_train_n_points hyperparameter, when I set regression_train_n_points equal to len(train_dataset) it gives me an error saying that the training dataset is too big, but yet it says in the error message that you can set this parameter equal to the length of the training dataset. That clearly is a bug that needs to be fixed, but when I specify a number lower than the length of the training dataset, it behaves in very weird ways throwing various warning signs about deprecated methods with pandas, 

one error message claiming that prophet needed 3 components claiming the dataset only had 1 component, which is false, the dataset had 84 components, 

and then I get an error message with AutoARIMA both created with parameters and without parameters saying this: 

ValueError: Train series only contains 21 elements but Auto-ARIMA model requires at least 30 entries

That does not make any sense, because first of all, why would it only take 21 out of 84 observations for training when I would've specified the length as 80% or 75% of the training data? I have no problems training the auto arima model when using model.fit(train), which tells me that this method is not behaving correctly and it needs to be looked at. 

Additionally, in the tutorial below, why is there 50 (or 12) observations designated for training, when normally you would just pass the entire training set to ensure a large enough sample for validation accuracy??? As of now I cannot use that approach per what I described earlier. 

**To Reproduce**
Here is a snippet of my code creating a workflow for testing various Regression methods using Sklearn:

```
for a in ['Ridge', 'Lasso', 'BayesianRidge', 'TweedieRegressor', 'SGDRegressor', 'LassoLars', 'SVM', 'GradientBoostingRegressor']:
model_eval_list=[]
    if a=='Ridge':
        Ridge=[]
        for b in np.arange(0, 1.1, .1):
            for c in [True, False]:
                model=RegressionEnsembleModel(forecasting_models=[m_prophet, m_arima, 
                                                        m_expon, m_bat, 
                                                                  m_tbat, m_fft], 
                                     regression_train_n_points=math.floor(len(train)*.75), 
                                      regression_model=sklearn.linear_model.Ridge(alpha=0.5, fit_intercept=True))
                accuracy=eval_model(model=model, train=train, val=val)
                Ridge.append((a, b, c, accuracy[1][0], 
                      accuracy[1][1], 
                      accuracy[1][2], 
                      accuracy[1][3], 
                      accuracy[1][4], 
                      accuracy[1][5], 
                      accuracy[1][6], 
                              accuracy[1][7]))
```

**Expected behavior**
What I am hoping to get out of this and the other issue I posted recently is to provide clear use cases for using this method beyond the following tutorial. If what I am doing is not conforming to a standardized way of doing this, then I would like to know what I am doing wrong so that I can adjust my code to fit that standard. But as of now, I cannot use Regression Ensemble model unless I have confidence that it works correctly. 

https://github.com/h3ik0th/Darts_ensembleFC/blob/main/darts16_update2022.01.ipynb

System:
 - Python version: 3.8
 - darts version 0.18.0

**Comment by hrzn** [comment]
> To start off with regression_train_n_points hyperparameter, when I set regression_train_n_points equal to len(train_dataset) it gives me an error saying that the training dataset is too big, but yet it says in the error message that you can set this parameter equal to the length of the training dataset. 

Could you copy past this error message, as well as a complete code snippet to reproduce this?

> one error message claiming that prophet needed 3 components claiming the dataset only had 1 component, which is false, the dataset had 84 components,

I'm not sure what you mean by component. Here too, can you please past a self-contained snippet to reproduce?

> That does not make any sense, because first of all, why would it only take 21 out of 84 observations for training when I would've specified the length as 80% or 75% of the training data? I have no problems training the auto arima model when using model.fit(train), which tells me that this method is not behaving correctly and it needs to be looked at.

`regression_train_n_points` is only used to train the stacker (the outer regression model). If this is too big, it won't leave enough training points to train the inner forecasting models (e.g., AutoARIMA). Try reducing `regression_train_n_points` to a small fraction of your train series length.

**Comment by hrzn** [comment]
This is a modelling question - hard to tell what is the good tradeoff in your case. It depends on your problem and data. You'll need to experiment to see what's right.

**Comment by GitDoctor1996** [comment]
> This is a modelling question - hard to tell what is the good tradeoff in your case. It depends on your problem and data. You'll need to experiment to see what's right.

Fair enough, but do you know of any examples to potentially explain the logic of the tradeoff? I understand the need to experiment, but I would like to understand the theoretical or mathematical reason for the potential tradeoff.
